### PR TITLE
feature(FR Numbers): add FR Numbers redesign

### PR DESCRIPTION
**Pull request by adi130987**

### CODE REVIEW

File: pmp/app-elements/interventions/intervention-fund-reservations.html [hunk 1]
```diff
@@ -158,86 +86,260 @@
...
-            if (dataItems.length) {
-              if (dataItems[0].value) {
-                return;
-              }
-              this.set('dataItems', dataItems.map(function(frNr) {
-                return {value: frNr};
-              }));
+        _frsUpdateWarningMessage: function(warning) {
+          if (this.frsConfirmationsDialog) {
+            let msgEl = Polymer.dom(this.frsConfirmationsDialog.root).querySelector('#frsConfirmationsDialogMessage');
```
Comment (adi130987) [inline pmp/app-elements/interventions/intervention-fund-reservations.html]: I'm gonna keep a private property that will contain this element and use it to update the message

File: pmp/app-elements/interventions/behaviors/fr-numbers-validations-behavior.html [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    compareFrsTotalAmountWithIntervData: function(frsDetails, intervention) {
+      let compareResult = null;
+      let pdDataNotMatched = [];
+      let frsDataNotMatched = [];
+
+      // check start dates
+      if (intervention.start !== frsDetails.earliest_start_date) {
+        pdDataNotMatched.push('Start Date (' + this.prettyDate(intervention.start) + ')');
+        frsDataNotMatched.push('FR Start Date (' + this.prettyDate(intervention.earliest_start_date) + ')');
+      }
```
Comment (acory) [inline pmp/app-elements/interventions/behaviors/fr-numbers-validations-behavior.html]: Can be encapsulated in a method and there will be no need for the comment. Same bellow

File: pmp/app-elements/interventions/fr-numbers/update-fr-numbers.html [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+        _checkFrNumbers: function(e) {
+          if (e.detail.confirmed) {
+            // resend rf number changes back to main fund reservations element
+            this.fire('update-frs-dialog-close', {frs: this._getUpdatedFrsNumbers()});
+          }
+        },
+
+        // prepare the rf numbers list, used to verify them using API endpoint (/api/v2/funds/frs)
+        _getUpdatedFrsNumbers: function() {
+          if (this.dataItems instanceof Array && this.dataItems.length > 0) {
```
Comment (acory) [inline pmp/app-elements/interventions/fr-numbers/update-fr-numbers.html]: Can be encapsulated in a method like nonEmptyArray(this.dataItems)

Comment (adi130987) [inline pmp/app-elements/interventions/fr-numbers/update-fr-numbers.html]: we will do this in the array-helper-behavior, we must use it all over the the app... future TODO

File: pmp/app-elements/interventions/behaviors/fr-numbers-validations-behavior.html (hunk 2 above)
Comment (adi130987) [inline pmp/app-elements/interventions/behaviors/fr-numbers-validations-behavior.html]: This ain't gonna be reusable.

**Pull request merged**